cmake/find/ccache.cmake

```diff
@@ -6,14 +6,19 @@ endif()
 
 if ((ENABLE_CCACHE OR NOT DEFINED ENABLE_CCACHE) AND NOT COMPILER_MATCHES_CCACHE)
     find_program (CCACHE_FOUND ccache)
+    if (CCACHE_FOUND)
+        set(ENABLE_CCACHE_BY_DEFAULT 1)
+    else()
+        set(ENABLE_CCACHE_BY_DEFAULT 0)
+    endif()
 endif()
 
 if (NOT CCACHE_FOUND AND NOT DEFINED ENABLE_CCACHE AND NOT COMPILER_MATCHES_CCACHE)
     message(WARNING "CCache is not found. We recommend setting it up if you build ClickHouse from source often. "
             "Setting it up will significantly reduce compilation time for 2nd and consequent builds")
 endif()
 
-option(ENABLE_CCACHE "Speedup re-compilations using ccache" ${CCACHE_FOUND})
```

Comment:
Seems like we cannot use random strings as default option values. This one previously evaluated to false even if ccache found.

Reply:
Looks like replacing `option` with `set(... CACHE ...)` should fix this (and this will not require extra if branch above)